### PR TITLE
Add `radical` icon (square root)

### PR DESCRIPTION
**Pull request by smnandre**
## What is the purpose of this pull request?
- [x] New Icon 
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description

Add a "square root" icon, in two formats like the other math operators

### Icon use case 

* Buttons for calculator
* Section/tags for sorting math lessons

### Concept 
- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license
- [x] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [x] I've based them on the following Lucide icons: activity, percent-square
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to two points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by jguddas** [comment]
The square-root feels a bit big.

**Comment by smnandre** [comment]
> The square-root feels a bit big.

On both icons ? Do you prefer like this ? 

square-root-smaller

**Comment by gruckion** [comment]
I like it

**Comment by smnandre** [comment]
Capture d’écran 2024-02-03 à 17 50 03
Capture d’écran 2024-02-03 à 17 46 39

File: icons/square-root-square.svg [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M19 3H5a2 2 0 0 0-2 2v14c0 1.1.9 2 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2Z"/>
+  <path d="M7 12h2l2 5 2-10h4"/>
```
Comment (jguddas) [inline icons/square-root-square.svg:13]:
```suggestion
  <path d="M7 12h2l2 5 2-10h4" />
  <rect x="3" y="3" width="18" height="18" rx="2" />
```

**Review by jguddas — CHANGES_REQUESTED** [review]

**Comment by jguddas** [comment]
And the name should be `radical` instead of `square-root`.

https://en.wikipedia.org/wiki/Radical_symbol

Besides the IMO better fitting name, it also doesn't mess with the group name `square`.

File: icons/square-root-square.svg (hunk 1 above)
Comment (smnandre) [reply icons/square-root-square.svg]: Thank you @jguddas

**Review by jguddas — APPROVED** [review]

**Comment by karsa-mistmere** [comment]
Nice work!

I think one final thing is in order: `radical-square` should be `square-radical` instead.
(Something that in the near future will be added to our naming guidelines).

**Comment by smnandre** [comment]
> Nice work!
Thank you (and all the feedback!)
 
> I think one final thing is in order: `radical-square` should be `square-radical` instead. (Something that in the near future will be added to our naming guidelines).

Ok! (i looked other icons, but said to myself it would be better if inversed, so... perfect :) )

**Review by karsa-mistmere — APPROVED** [review]

**Comment by smnandre** [comment]
Not sure to understand the linter message

> Contributors have changed!
> 
> Don't add people who have only performed automatic optimizations.
> 
> +}

Is there something i should do there ?

**Review by jguddas — APPROVED** [review]

**Review by ericfennis — APPROVED** [review]

**Comment by danielbayley** [comment]
@smnandre Nice, I have almost identical in my drafts… Should plain `radical` not also be horizontally centred though?

> I think one final thing is in order: `radical-square` should be `square-radical` instead. (Something that in the near future will be added to our naming guidelines).

@karsa-mistmere My line of thinking with previous naming was the other way around… The reasoning being that in a sorted list, having to search through _many_ `square-*` icons would be much less user friendly than having them more grouped by `name-square`… What do you think? cc: @ericfennis @jguddas

**Comment by smnandre** [comment]
> @smnandre Nice, I have almost identical in my drafts… Should plain `radical` not also be horizontally centred though?

It was an attempt to improve a bit the readability of the shape... all my respect for spotting it :) 

Capture d’écran 2024-02-10 à 03 39 39


Before(svg) before 
After(svg) after 

So you prefer the centered one ? @danielbayley 

For the name tell me, and i'll change the names :)

**Comment by karsa-mistmere** [comment]
I've been thinking that possibly we should also consider consistency with `activity` and keep the lower end 1px below the base line, this would also improve vertical centering. 🤔
image
image

**Comment by smnandre** [comment]
> I've been thinking that possibly we should also consider consistency with `activity` and keep the lower end 1px below the base line, this would also improve vertical centering. 🤔

I tried to be as aligned as possible with the other mathematical symbols, the ones that probably would be near this one: cf https://github.com/lucide-icons/lucide/pull/1847#issuecomment-1925394067

**Comment by danielbayley** [comment]
> For the name tell me, and i'll change the names :)

The order needs to be agreed, and codified into the guidelines, as @karsa-mistmere mentioned…

**Comment by karsa-mistmere** [comment]
> The order needs to be agreed, and codified into the guidelines, as @karsa-mistmere mentioned…

I'm pretty sure, @ericfennis, @jguddas and I have already agreed upon `square-*` vs. `*-square`, we just haven't had time to put it in writing yet. So `square-radical` is definitely the way to go.

**Comment by smnandre** [comment]
Ok so i don't have to re-rename them! Tell me if you need anything else :)

**Review by jguddas — APPROVED** [review]

File: icons/radical.svg [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  width="24"
+  height="24"
+  viewBox="0 0 24 24"
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path d="M3 12h4l3 8 3-16h7" />
```
Comment (karsa-mistmere) [inline icons/radical.svg:12]: I think we should add an extra pixel of height to the bottom both to better match `activity` and to adjust the centre of gravity vertically:
```suggestion
  <path d="M3 12h4l3 9 4-17h7" />
```

**Comment by danielbayley** [comment]
> I'm pretty sure, @ericfennis, @jguddas and I have already agreed upon `square-*` vs. `*-square`

@karsa-mistmere Ah ok, where can I catch up on the reasoning, was there an issue?

**Comment by jguddas** [comment]
> > I'm pretty sure, @ericfennis, @jguddas and I have already agreed upon `square-*` vs. `*-square`
> 
> @karsa-mistmere Ah ok, where can I catch up on the reasoning, was there an issue?

#1874

**Comment by karsa-mistmere** [comment]
> @karsa-mistmere Ah ok, where can I catch up on the reasoning, was there an issue?

Originally, there was some discussion in https://github.com/lucide-icons/lucide/pull/1638 regarding this matter, most of the reasoning for why we went with what we did is there, on Discord we just formalized this decision into [some kind of rule](https://github.com/lucide-icons/lucide/pull/1874) that would be easy to follow by our contributors and hopefully result in reasonably consistent names.

**Comment by smnandre** [comment]
Is there something more i can do on this one ? :)

**Review by karsa-mistmere — APPROVED** [review]

**Pull request merged**